<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SettingsDialog</class>
 <widget class="QDialog" name="SettingsDialog">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1358</width>
    <height>902</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Settings</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_3">
   <item row="0" column="0">
    <widget class="QTabWidget" name="settings">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>General</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Number of CPU processes:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="num_cpu">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>256</number>
         </property>
         <property name="value">
          <number>10</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Number of GPU processes:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="num_gpu">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>256</number>
         </property>
         <property name="value">
          <number>2</number>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string extracomment="comma separated list of IDs">GPU IDs:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="gpu_ids">
         <property name="text">
          <string>0,1</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Process timeout:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QSpinBox" name="timeout">
         <property name="suffix">
          <string>s</string>
         </property>
         <property name="maximum">
          <number>9999999</number>
         </property>
         <property name="value">
          <number>300</number>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>Number of bins for histogram:</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QSpinBox" name="histogram_bins">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>256</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Tasks</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>Tasks</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QTreeWidget" name="task_tree">
            <property name="dragEnabled">
             <bool>true</bool>
            </property>
            <property name="defaultDropAction">
             <enum>Qt::MoveAction</enum>
            </property>
            <property name="itemsExpandable">
             <bool>false</bool>
            </property>
            <property name="expandsOnDoubleClick">
             <bool>false</bool>
            </property>
            <property name="columnCount">
             <number>4</number>
            </property>
            <attribute name="headerVisible">
             <bool>false</bool>
            </attribute>
            <attribute name="headerCascadingSectionResizes">
             <bool>false</bool>
            </attribute>
            <attribute name="headerHighlightSections">
             <bool>false</bool>
            </attribute>
            <attribute name="headerStretchLastSection">
             <bool>false</bool>
            </attribute>
            <column>
             <property name="text">
              <string>Name</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Script</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>GPU task</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Group with parent</string>
             </property>
            </column>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Input variables</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QTableWidget" name="input_variable_table">
            <column>
             <property name="text">
              <string>Display name</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Variable</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Type</string>
             </property>
            </column>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Outputs variables</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QTableWidget" name="output_variable_table">
            <property name="selectionMode">
             <enum>QAbstractItemView::SingleSelection</enum>
            </property>
            <property name="selectionBehavior">
             <enum>QAbstractItemView::SelectRows</enum>
            </property>
            <column>
             <property name="text">
              <string>Display Name</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Variable</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Type</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Display in column</string>
             </property>
            </column>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Export</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="5" column="0">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Number of processes:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="export_num_processes">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>1</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="4" column="0" colspan="2">
        <widget class="QGroupBox" name="groupBox_5">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Scripts</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="label_5">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Custom export script:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="PathEdit" name="export_custom_script" native="true">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Pre export script:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Post export script:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="PathEdit" name="export_pre_script" native="true"/>
          </item>
          <item row="2" column="1">
           <widget class="PathEdit" name="export_post_script" native="true"/>
          </item>
          <item row="3" column="0">
           <widget class="QCheckBox" name="ask_destination">
            <property name="text">
             <string>Ask for destination directory</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="1" rowspan="2">
        <widget class="QGroupBox" name="groupBox_4">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Export type</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QRadioButton" name="export_copy">
            <property name="text">
             <string>Copy data</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="export_move">
            <property name="text">
             <string>&amp;Move data</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="export_custom1">
            <property name="text">
             <string>Use custom script per image</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="export_custom2">
            <property name="text">
             <string>Use global custom script </string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="1" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QPushButton" name="load">
       <property name="text">
        <string>Load...</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="save">
       <property name="text">
        <string>Save...</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="save_defaults">
       <property name="text">
        <string>Save as defaults</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="reset_to_defaults">
       <property name="text">
        <string>Reset to defaults</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
       </property>
       <property name="centerButtons">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PathEdit</class>
   <extends>QWidget</extends>
   <header>pathedit.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>601</x>
     <y>891</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>669</x>
     <y>891</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>load</sender>
   <signal>clicked()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>loadFromFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>63</x>
     <y>891</y>
    </hint>
    <hint type="destinationlabel">
     <x>4</x>
     <y>390</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>save</sender>
   <signal>clicked()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>saveToFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>175</x>
     <y>891</y>
    </hint>
    <hint type="destinationlabel">
     <x>117</x>
     <y>385</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>export_custom2</sender>
   <signal>toggled(bool)</signal>
   <receiver>groupBox_5</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>776</x>
     <y>171</y>
    </hint>
    <hint type="destinationlabel">
     <x>783</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>export_custom1</sender>
   <signal>toggled(bool)</signal>
   <receiver>groupBox_5</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>905</x>
     <y>142</y>
    </hint>
    <hint type="destinationlabel">
     <x>897</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>export_custom1</sender>
   <signal>toggled(bool)</signal>
   <receiver>export_num_processes</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>974</x>
     <y>143</y>
    </hint>
    <hint type="destinationlabel">
     <x>534</x>
     <y>85</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>save_defaults</sender>
   <signal>clicked()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>saveAsDefaults()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>251</x>
     <y>874</y>
    </hint>
    <hint type="destinationlabel">
     <x>3</x>
     <y>350</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>reset_to_defaults</sender>
   <signal>clicked()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>resetToDefaults()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>324</x>
     <y>879</y>
    </hint>
    <hint type="destinationlabel">
     <x>293</x>
     <y>897</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>loadFromFile()</slot>
  <slot>saveToFile()</slot>
  <slot>saveAsDefaults()</slot>
  <slot>resetToDefaults()</slot>
 </slots>
</ui>
